zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/ProcessDefinitionQueryControllerTest.java

```diff
@@ -190,6 +192,34 @@ public void shouldReturnProcessDefinitionForValidKey() {
     verify(processDefinitionServices).getByKey(23L);
   }
 
+  @Test
+  public void shouldReturn403ForForbiddenProcessDefinitionKey() {
```

Comment:
🔧 Can we add test cases for the form and XML endpoints as well to ensure they use the same permission checks? That makes it simpler to understand later which endpoints use which permissions by just looking at the test cases.